server/controllers/profile.js

```diff
@@ -0,0 +1,69 @@
+const mongoose = require('mongoose');
+const Profile = require('../models/Profile');
+const asyncHandler = require('express-async-handler');
+
+// POST /profile - Create new profile
+exports.createNewProfile = asyncHandler(async (req, res) => {
+  const profile = req.body;
+  const newProfile = new Profile({ ...profile });
+
+  if (!newProfile) {
+    res.status(500);
+    throw new Error('Error creating the profile.');
+  }
+  await newProfile.save();
+  res.status(201).json({ status: 201, profile: newProfile, message: 'Profile created!' });
+});
+
+// UPDATE /profile:id - Update profile with the given id
+exports.updateProfile = asyncHandler(async (req, res) => {
+  const id = req.params.id;
```

Comment:
Since we will be updating the profile for logged in users, you can get the logged in user's id from the request object here.

Hint: make sure you add the `auth` middleware to your routes

Reply:
Can you elaborate what you meant by this? 

In terms of using auth middleware to the routes, is this what you meant?
![authmiddleware](https://user-images.githubusercontent.com/62577683/129242265-8dc78628-06a5-42e9-96bc-4e133517851b.jpg)

Reply:
Yes, that is correct!